articles/introduction-to-avl-trees/index.md

```diff
@@ -0,0 +1,548 @@
+        else if(val < node->value)
+            node->leftChild = delete_AVL(node->leftChild, val);
+        else if(val > node->value)
+            node->rightChild = delete_AVL(node->rightChild, val);
+
+        //Element found with two children
+        else if(node->leftChild && node->rightChild)
+        {
+            temp = findMin(node->rightChild);
+            node->value = temp->value;
+            node->rightChild = delete_AVL(node->rightChild, node->value);
+        }
+
+        //Element with one or zero child
+        else
+        {
+            temp = node;
+            if(node->leftChild == NULL)
+                node = node->rightChild;
+            else if(node->rightChild == NULL)
+                node = node->leftChild;
+            delete temp;
+        }
+
+        //If the left node is deleted, right case
+        if(getBalance(node) == 2)
+        {
+            //right-right case
+            if(getBalance(node->leftChild) == 1)
+                return rr_rotate(node);
+            //right-left case
+            else
+                return rl_rotate(node);
+        }
+        //If right node is deleted, left case
+        else if(getBalance(node) == -2)
+        {
+            //left-left case
+            if(getBalance(node->rightChild)== 1)
+                return ll_rotate(node);
+            //left-right case
+            else
+                return lr_rotate(node);
+        }
+        return node;
+}
+
+void inOrder(Node *root)  
+{  
+    if(root != NULL)  
+    {  
+        inOrder(root->leftChild); 
+        cout << root->value << " ";  
+        inOrder(root->rightChild);  
+    }  
+} 
+//Driver Code
+int main(void)
+{
+    Node *root = NULL;  
+    root = insert_AVL(root, 1);  
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    root = insert_AVL(root, 3);
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    root = insert_AVL(root, 4); 
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    root = insert_AVL(root, 5); 
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    root = insert_AVL(root, 6);
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    root = insert_AVL(root, 2);  
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);  
+    cout<<endl;
+    root = delete_AVL(root, 2);  
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    root = delete_AVL(root, 6);  
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+       
+}
+```
+
+The time complexity of deletion is O(logn) as well. This is due to the self-balancing feature of the tree. 
+
```

Comment:
A "Further Reading" section with articles related to C++ and AVL Trees would complement your article well.

Example, https://www.section.io/engineering-education/validating-xml-using-dtd/#further-reading

Reply:
Revisions made in the recent commit.